bitnami/mariadb/values-production.yaml

```diff
@@ -116,6 +116,9 @@ secret:
   ## Add custom annotations to the secret (evaluated as a template)
   ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
   annotations: {}
+  ## Require all passwords at the secret resource
+  ##
+  requirePasswords: true
```

Comment:
I wouldn't add this. Having `requirePassword` per each password should be enough. If you want or need it you can calculate it like:

```
requirePasswords = db.password.requirePassword || db.rootUser.requirePassword || replication.requirePassword
```

Reply:
If I had to choose between `secret.requirePasswords` and `db.password.requirePassword`,  `db.rootUser.requirePassword`, `replication.requirePassword`, I prefer `secret.requirePasswords`:

1. It's more common auto generate all passwords, not only one.
2. Less params.
```yml
secret:
  requirePasswords: false
  annotations:
    secret-generator.v1.mittwald.de/autogenerate: mariadb-root-password,mariadb-replication-password,mariadb-password
```

```yml
rootUser:
  requirePassword: false
db:
  requirePassword: false
replication:
  requirePassword: false
secret:
  annotations:
    secret-generator.v1.mittwald.de/autogenerate: mariadb-root-password,mariadb-replication-password,mariadb-password
```
3. With `db.password.requirePassword`, `db.rootUser.requirePassword` or`replication.requirePassword` there is a problem.
If secret resource has the [data](https://github.com/bitnami/charts/pull/3411/files#diff-2b1369100f6cef406924950c2e18e566R17) key will cause a helm upgrade conflict

If secret resource has the data key, it will causes a helm upgrade conflict.

Reply:
let's go for `secret.requirePasswords` do you agree?

Reply:
👍